FRAuth/FRAuth/Network/FRHTTPCookie.swift

```diff
@@ -0,0 +1,88 @@
+// 
+//  FRHTTPCookie.swift
+//  FRAuth
+//
+//  Copyright (c) 2021 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+
+class FRHTTPCookie: HTTPCookie, NSSecureCoding {
+    required init?(coder: NSCoder) {
+        var properties = [HTTPCookiePropertyKey: Any]()
+        let version = coder.decodeInteger(forKey: "version")
+        let name = coder.decodeObject(of: NSString.self, forKey: "name") as String?
+        let value = coder.decodeObject(of: NSString.self, forKey: "value") as String?
+        let expiresDate = coder.decodeObject(of: NSDate.self, forKey: "expiresDate") as Date?
+        let isSessionOnly = coder.decodeBool(forKey: "isSessionOnly")
+        let domain = coder.decodeObject(of: NSString.self, forKey: "domain") as String?
+        let path = coder.decodeObject(of: NSString.self, forKey: "path") as String?
+        let isSecure = coder.decodeBool(forKey: "isSecure")
+        let isHTTPOnly = coder.decodeBool(forKey: "isHTTPOnly")
+        
+        let comment = coder.decodeObject(of: NSString.self, forKey: "comment") as String?
+        let commentURL = coder.decodeObject(of: NSURL.self, forKey: "commentURL") as URL?
+        let portList: [NSNumber]?
+        if #available(iOS 14.0, *) {
+            portList = coder.decodeArrayOfObjects(ofClass: NSNumber.self, forKey: "portList")
+        } else {
+            portList = coder.decodeObject(of: [NSNumber.self], forKey: "portList") as? [NSNumber]
+        }
+        
+        let sameSitePolicy = coder.decodeObject(of: NSString.self, forKey: "sameSitePolicy") as String?
+        
+        properties[HTTPCookiePropertyKey.version] = version
+        properties[HTTPCookiePropertyKey.name] = name
+        properties[HTTPCookiePropertyKey.value] = value
+        properties[HTTPCookiePropertyKey.domain] = domain
+        properties[HTTPCookiePropertyKey.path] = path
+        properties[HTTPCookiePropertyKey.secure] = isSecure
+        properties[HTTPCookiePropertyKey.expires] = expiresDate
+        properties[HTTPCookiePropertyKey.comment] = comment
+        properties[HTTPCookiePropertyKey.commentURL] = commentURL
+        properties[HTTPCookiePropertyKey.discard] = isSessionOnly
+        properties[HTTPCookiePropertyKey.maximumAge] = expiresDate
+        properties[HTTPCookiePropertyKey.port] = portList
+        properties[HTTPCookiePropertyKey("HttpOnly")] = isHTTPOnly
+        if #available(iOS 13.0, *) {
```

Comment:
Same as above comment

Reply:
This class is only used in the iOS 11+ scenario. But this is a good shout as I might need to specify some compiler flags here. Will have a look.

iOS 10 will always use HTTPCookie instead of the FRHTTPCookie